Game/ConsoleApplication2/MovingGameObject.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include "GameObject.h"
+#include "Constants.h"
+
+
+class MovingGameObject : public GameObject
+{
```

Comment:
Not sure about the MovingGameObject class being necessary, I understand you don't want to duplicate code, but this is definitely not how you do it. Keep this logic in the astronaut class, and only take it out of there when we need to, we can't think that far ahead, so don't code like that. AIObject being a class I am okay with, and all that does is call static functions in the AIManager, that makes sense and is less use case specific and scalable to more objects. Argue with this point if you feel its necessary.